### PR TITLE
[Feature] jdbc sql server draft

### LINKED ISSUE #30704: 请求JDBC CATALOG中增加oracle和sqlserver数据源类型
- [x] https://github.com/StarRocks/starrocks/pull/46444
JDBC CATALOG中增加oracle和sqlserver数据源类型

### PR DESCRIPTION
**Pull request by XinzhuangL**
## What I'm doing:

Fixes #30704

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/connector/jdbc/SqlServerSchemaResolver.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    @Override
+    public Table getTable(long id, String name, List<Column> schema, List<Column> partitionColumns, String dbName,
+                          String catalogName, Map<String, String> properties) throws DdlException {
+        Map<String, String> newProp = new HashMap<>(properties);
+        newProp.putIfAbsent(JDBCTable.JDBC_TABLENAME, "\"" + dbName + "\"" + "." + "\"" + name + "\"");
+        return new JDBCTable(id, name, schema, partitionColumns, dbName, catalogName, newProp);
+    }
+
+    @Override
+    public Type convertColumnType(int dataType, String typeName, int columnSize, int digits) {
```
Comment (zombee0) [inline fe/fe-core/src/main/java/com/starrocks/connector/jdbc/SqlServerSchemaResolver.java:81]: nice job!

File: fe/fe-core/src/test/java/com/starrocks/connector/jdbc/SqlServerSchemaResolverTest.java [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+            }
+        };
+        try {
+            JDBCMetadata jdbcMetadata = new JDBCMetadata(properties, "catalog", dataSource);
+            Table table = jdbcMetadata.getTable("test", "tbl1");
+            Assert.assertTrue(table instanceof JDBCTable);
+            Assert.assertEquals("catalog.test.tbl1", table.getUUID());
+            Assert.assertEquals("tbl1", table.getName());
+            Assert.assertNull(properties.get(JDBCTable.JDBC_TABLENAME));
+            SqlServerSchemaResolver sqlServerSchemaResolver = new SqlServerSchemaResolver();
```
[the remark this answers is not in the document]
Comment (XinzhuangL) [reply fe/fe-core/src/test/java/com/starrocks/connector/jdbc/SqlServerSchemaResolverTest.java]: I have modified it according to the latest code

**Review by zombee0 — APPROVED** [review]
LGTM

File: fe/pom.xml [hunk 3]
```diff
@@ -810,6 +810,13 @@ under the License.
                 <version>19.18.0.0</version>
             </dependency>
 
+            <!-- https://mvnrepository.com/artifact/com.microsoft.sqlserver/mssql-jdbc/12.4.2.jre11 -->
+            <dependency>
+                <groupId>com.microsoft.sqlserver</groupId>
+                <artifactId>mssql-jdbc</artifactId>
+                <version>12.4.2.jre11</version>
```
Comment (Smith-Cruise) [inline fe/pom.xml:817]: plz using https://aquasecurity.github.io/trivy/v0.52/ tools to check this dependency will introduce cves.

Comment (XinzhuangL) [reply fe/pom.xml:817]: Thank you very much for your advice. I checked the pom files, and the high risks all seem to be related to trino

Total: 23 (UNKNOWN: 0, LOW: 1, MEDIUM: 12, HIGH: 10, CRITICAL: 0)

┌───────────────────────────────────────────┬────────────────┬──────────┬──────────┬───────────────────┬───────────────┬──────────────────────────────────────────────────────────────┐
│                  Library                  │ Vulnerability  │ Severity │  Status  │ Installed Version │ Fixed Version │                            Title                             │
├───────────────────────────────────────────┼────────────────┼──────────┼──────────┼───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ com.nimbusds:nimbus-jose-jwt              │ CVE-2023-52428 │ MEDIUM   │ fixed    │ 9.31              │ 9.37.2        │ Denial of Service in Connect2id Nimbus JOSE+JWT              │
│                                           │                │          │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-52428                   │
├───────────────────────────────────────────┼────────────────┤          │          ├───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ com.squareup.okio:okio-jvm                │ CVE-2023-3635  │          │          │ 3.0.0             │ 3.4.0         │ okio: GzipSource class improper exception handling           │
│                                           │                │          │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-3635                    │
├───────────────────────────────────────────┼────────────────┼──────────┤          ├───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ io.airlift:aircompressor                  │ CVE-2024-36114 │ HIGH     │          │ 0.10              │ 0.27          │ Decompressors can crash the JVM and leak memory content in   │
│                                           │                │          │          │                   │               │ Aircompressor                                                │
│                                           │                │          │          │                   │               │ https://avd.aquasec.com/nvd/cve-2024-36114                   │
│                                           │                │          │          ├───────────────────┤               │                                                              │
│                                           │                │          │          │ 0.26              │               │                                                              │
│                                           │                │          │          │                   │               │                                                              │
│                                           │                │          │          │                   │               │                                                              │
├───────────────────────────────────────────┼────────────────┼──────────┤          ├───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ io.netty:netty-codec-http                 │ CVE-2024-29025 │ MEDIUM   │          │ 4.1.100.Final     │ 4.1.108.Final │ netty-codec-http: Allocation of Resources Without Limits or  │
│                                           │                │          │          │                   │               │ Throttling                                                   │
│                                           │                │          │          │                   │               │ https://avd.aquasec.com/nvd/cve-2024-29025                   │
├───────────────────────────────────────────┼────────────────┼──────────┤          ├───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ org.apache.avro:avro                      │ CVE-2023-39410 │ HIGH     │          │ 1.9.2             │ 1.11.3        │ apache-avro: Apache Avro Java SDK: Memory when deserializing │
│                                           │                │          │          │                   │               │ untrusted data in Avro...                                    │
│                                           │                │          │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-39410                   │
├───────────────────────────────────────────┼────────────────┤          │          ├───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ org.apache.commons:commons-compress       │ CVE-2024-25710 │          │          │ 1.22              │ 1.26.0        │ commons-compress: Denial of service caused by an infinite    │
│                                           │                │          │          │                   │               │ loop for a corrupted...                                      │
│                                           │                │          │          │                   │               │ https://avd.aquasec.com/nvd/cve-2024-25710                   │
│                                           ├────────────────┤          │          │                   │               ├──────────────────────────────────────────────────────────────┤
│                                           │ CVE-2024-26308 │          │          │                   │               │ commons-compress: OutOfMemoryError unpacking broken Pack200  │
│                                           │                │          │          │                   │               │ file                                                         │
│                                           │                │          │          │                   │               │ https://avd.aquasec.com/nvd/cve-2024-26308                   │
│                                           ├────────────────┼──────────┤          │                   ├───────────────┼──────────────────────────────────────────────────────────────┤
│                                           │ CVE-2023-42503 │ MEDIUM   │          │                   │ 1.24.0        │ Improper Input Validation, Uncontrolled Resource Consumption │
│                                           │                │          │          │                   │               │ vulnerabi ...                                                │
│                                           │                │          │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-42503                   │
│                                           ├────────────────┼──────────┤          ├───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│                                           │ CVE-2024-25710 │ HIGH     │          │ 1.24.0            │ 1.26.0        │ commons-compress: Denial of service caused by an infinite    │
│                                           │                │          │          │                   │               │ loop for a corrupted...                                      │
│                                           │                │          │          │                   │               │ https://avd.aquasec.com/nvd/cve-2024-25710                   │
│                                           ├────────────────┤          │          │                   │               ├──────────────────────────────────────────────────────────────┤
│                                           │ CVE-2024-26308 │          │          │                   │               │ commons-compress: OutOfMemoryError unpacking broken Pack200  │
│                                           │                │          │          │                   │               │ file                                                         │
│                                           │                │          │          │                   │               │ https://avd.aquasec.com/nvd/cve-2024-26308                   │
├───────────────────────────────────────────┼────────────────┼──────────┤          ├───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ org.apache.commons:commons-configuration2 │ CVE-2024-29131 │ MEDIUM   │          │ 2.8.0             │ 2.10.1        │ commons-configuration: StackOverflowError adding property in │
│                                           │                │          │          │                   │               │ AbstractListDelimiterHandler.flattenIterator()               │
│                                           │                │          │          │                   │               │ https://avd.aquasec.com/nvd/cve-2024-29131                   │
│                                           ├────────────────┤          │          │                   │               ├──────────────────────────────────────────────────────────────┤
│                                           │ CVE-2024-29133 │          │          │                   │               │ commons-configuration: StackOverflowError calling            │
│                                           │                │          │          │                   │               │ ListDelimiterHandler.flatten(Object, int) with a cyclical    │
│                                           │                │          │          │                   │               │ object tree                                                  │
│                                           │                │          │          │                   │               │ https://avd.aquasec.com/nvd/cve-2024-29133                   │
├───────────────────────────────────────────┼────────────────┼──────────┤          ├───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ org.apache.solr:solr-solrj                │ CVE-2023-50298 │ LOW      │          │ 8.11.2            │ 9.4.1, 8.11.3 │ solr: possible exposure of ZooKeeper credentials via         │
│                                           │                │          │          │                   │               │ Streaming Expressions                                        │
│                                           │                │          │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-50298                   │
├───────────────────────────────────────────┼────────────────┼──────────┼──────────┼───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ org.bouncycastle:bcprov-jdk15on           │ CVE-2023-33201 │ MEDIUM   │ affected │ 1.70              │               │ bouncycastle: potential blind LDAP injection attack using a  │
│                                           │                │          │          │                   │               │ self-signed certificate                                      │
│                                           │                │          │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-33201                   │
│                                           ├────────────────┤          ├──────────┤                   ├───────────────┼──────────────────────────────────────────────────────────────┤
│                                           │ CVE-2024-29857 │          │ fixed    │                   │ 1.78          │ An issue was discovered in ECCurve.java and ECCurve.cs in    │
│                                           │                │          │          │                   │               │ Bouncy Castl ......                                          │
│                                           │                │          │          │                   │               │ https://avd.aquasec.com/nvd/cve-2024-29857                   │
│                                           ├────────────────┤          │          │                   │               ├──────────────────────────────────────────────────────────────┤
│                                           │ CVE-2024-30171 │          │          │                   │               │ bc-java: BouncyCastle vulnerable to a timing variant of      │
│                                           │                │          │          │                   │               │ Bleichenbacher (Marvin Attack)                               │
│                                           │                │          │          │                   │               │ https://avd.aquasec.com/nvd/cve-2024-30171                   │
│                                           ├────────────────┤          │          │                   │               ├──────────────────────────────────────────────────────────────┤
│                                           │ CVE-2024-30172 │          │          │                   │               │ An issue was discovered in Bouncy Castle Java Cryptography   │
│                                           │                │          │          │                   │               │ APIs before ......                                           │
│                                           │                │          │          │                   │               │ https://avd.aquasec.com/nvd/cve-2024-30172                   │
├───────────────────────────────────────────┼────────────────┼──────────┤          ├───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ org.jboss.xnio:xnio-api                   │ CVE-2023-5685  │ HIGH     │          │ 3.8.10.Final      │ 3.8.14.Final  │ xnio: StackOverflowException when the chain of notifier      │
│                                           │                │          │          │                   │               │ states becomes problematically big                           │
│                                           │                │          │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-5685                    │
├───────────────────────────────────────────┼────────────────┤          │          ├───────────────────┼───────────────┼──────────────────────────────────────────────────────────────┤
│ org.xerial.snappy:snappy-java             │ CVE-2023-34455 │          │          │ 1.1.8.3           │ 1.1.10.1      │ snappy-java: Unchecked chunk length leads to DoS             │
│                                           │                │          │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-34455                   │
│                                           ├────────────────┤          │          │                   ├───────────────┼──────────────────────────────────────────────────────────────┤
│                                           │ CVE-2023-43642 │          │          │                   │ 1.1.10.4      │ snappy-java: Missing upper bound check on chunk length in    │
│                                           │                │          │          │                   │               │ snappy-java can lead...                                      │
│                                           │                │          │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-43642                   │
│                                           ├────────────────┼──────────┤          │                   ├───────────────┼──────────────────────────────────────────────────────────────┤
│                                           │ CVE-2023-34453 │ MEDIUM   │          │                   │ 1.1.10.1      │ snappy-java: Integer overflow in shuffle leads to DoS        │
│                                           │                │          │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-34453                   │
│                                           ├────────────────┤          │          │                   │               ├──────────────────────────────────────────────────────────────┤
│                                           │ CVE-2023-34454 │          │          │                   │               │ snappy-java: Integer overflow in compress leads to DoS       │
│                                           │                │          │          │                   │               │ https://avd.aquasec.com/nvd/cve-2023-34454                   │
└───────────────────────────────────────────┴────────────────┴──────────┴──────────┴───────────────────┴───────────────┴──────────────────────────────────────────────────────────────┘

Comment (XinzhuangL) [reply fe/pom.xml:817]: 👍🏻 I re-executed the trivy scan and its behavior is consistent with that of the main branch, did not contain critical errors, but contains high. miedium and low. I'm not familiar with trivy, I don't know if it's normal. But my submission doesn't seem to introduce new cves.

Comment (Smith-Cruise) [reply fe/pom.xml:817]: ok, got it.

**Review by DorianZheng — APPROVED** [review]

**Pull request merged**